numba/listobject.py

```diff
@@ -1083,6 +1170,13 @@ def compare(this, other):
     """
     if len(this) != len(other):
         return -1 if len(this) < len(other) else 1
+    if this_is_none or other_is_none:
+        if this_is_none and other_is_none:
+            return 0
+        elif this_is_none and not other_is_none:
+            return -1
+        elif not this_is_none and other_is_none:
+            return 1
```

Comment:
I think some of this may be superfluous... is this logically equivalent?
```python
    if this_is_none or other_is_none: # one of them is none
        if this_is_none and other_is_none: # both none
            return 0
        # to get here there is precisely one none, and if the first is none, by induction, the second cannot be?
        return -1 if this_is_none else 1
```

Reply:
yes it is

Reply:
done